### PR TITLE
TAS: Add integration tests for job.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add TAS integration tests for job.

#### Which issue(s) this PR fixes:

Part of #3372

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/controller/jobs/job/job_controller_test.go [hunk 1]
```diff
@@ -2158,6 +2161,238 @@ var _ = ginkgo.Describe("Job controller interacting with Workload controller whe
...
+		features.SetFeatureGateDuringTest(ginkgo.GinkgoTB(), features.TopologyAwareScheduling, true)
+
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "tas-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		nodes = []corev1.Node{
```
Comment (mimowo) [inline test/integration/controller/jobs/job/job_controller_test.go:2197]: Let's simplify the test maximally, we just want to play it a role of "sanity check". We have separate tests for more complex structures based on workloads. 

So, let's have only one node with one topology level (hostname).

File: test/integration/controller/jobs/job/job_controller_test.go [hunk 2]
```diff
@@ -2158,6 +2161,238 @@ var _ = ginkgo.Describe("Job controller interacting with Workload controller whe
...
+					&kueue.TopologyAssignment{
+						Levels:  []string{tasBlockLabel, tasRackLabel},
+						Domains: []kueue.TopologyDomainAssignment{{Count: 1, Values: []string{"b1", "r1"}}},
+					},
+				))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+	})
+
+	ginkgo.It("should admit workload which fits in a preferred topology domain", func() {
```
Comment (mimowo) [inline test/integration/controller/jobs/job/job_controller_test.go:2347]: I think we can drop the test entirely. Let's make the intergration tests per CRD minimal "sanity checks". We can have more rigorous tests using workloads.

File: test/integration/controller/jobs/job/suite_test.go [hunk 3]
```diff
@@ -85,30 +86,29 @@ func managerSetup(opts ...jobframework.Option) framework.ManagerSetup {
...
 
-		err = job.SetupIndexes(ctx, mgr.GetFieldIndexer())
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		err = job.NewReconciler(mgr.GetClient(),
-			mgr.GetEventRecorderFor(constants.JobControllerName), opts...).SetupWithManager(mgr)
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		err = job.SetupWebhook(mgr, opts...)
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		jobframework.EnableIntegration(job.FrameworkName)
+		if setupTAS {
```
Comment (mimowo) [inline test/integration/controller/jobs/job/suite_test.go:108]:
```suggestion
		if setupTASControllers {
```

File: test/integration/controller/jobs/job/job_controller_test.go [hunk 4]
```diff
@@ -2158,6 +2161,238 @@ var _ = ginkgo.Describe("Job controller interacting with Workload controller whe
...
+	ginkgo.AfterAll(func() {
+		fwk.StopManager(ctx)
+	})
+
+	ginkgo.BeforeEach(func() {
+		features.SetFeatureGateDuringTest(ginkgo.GinkgoTB(), features.TopologyAwareScheduling, true)
+
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "tas-",
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply test/integration/controller/jobs/job/job_controller_test.go]: Done

File: test/integration/controller/jobs/job/suite_test.go (hunk 3 above)
Comment (mbobrovskyi) [reply test/integration/controller/jobs/job/suite_test.go]: Done

File: test/integration/controller/jobs/job/job_controller_test.go (hunk 2 above)
Comment (mbobrovskyi) [reply test/integration/controller/jobs/job/job_controller_test.go]: Done

File: test/integration/controller/jobs/job/job_controller_test.go (hunk 1 above)
Comment (mbobrovskyi) [reply test/integration/controller/jobs/job/job_controller_test.go:2197]: Done

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
/hold 
in case @tenzen-y has more comments

**Comment by tenzen-y** [comment]
> /lgtm /approve /hold in case @tenzen-y has more comments

ACK

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/approve

**Pull request merged**